t/core/ctx2.t

```diff
@@ -136,3 +136,69 @@ qr/balancer_ip: 127.0.0.1|balancer_port: 1980/
 --- grep_error_log_out
 balancer_ip: 127.0.0.1
 balancer_port: 1980
+
+
+
+=== TEST 5: support dash in the args
+--- config
+    location /t {
+        content_by_lua_block {
+            local core = require("apisix.core")
+            local ngx_ctx = ngx.ctx
+            local api_ctx = ngx_ctx.api_ctx
+            if api_ctx == nil then
+                api_ctx = core.tablepool.fetch("api_ctx", 0, 32)
```

Comment:
It is not a good idea to set up api_ctx directly. We should let APISIX set it. See https://github.com/apache/apisix/pull/4676#discussion_r680302823